drivers/sensor/icm42605/icm42605.c

```diff
@@ -0,0 +1,464 @@
+			     + (drv_data->fifo_data[2] & 0xff);
+		result = inv_spi_read(REG_FIFO_DATA, drv_data->fifo_data,
+				      fifo_count);
+
+		/* FIFO Data structure
+		 * Packet 1 : FIFO Header(1), AccelX(2), AccelY(2),
+		 *            AccelZ(2), Temperature(1)
+		 * Packet 2 : FIFO Header(1), GyroX(2), GyroY(2),
+		 *            GyroZ(2), Temperature(1)
+		 * Packet 3 : FIFO Header(1), AccelX(2), AccelY(2), AccelZ(2),
+		 *            GyroX(2), GyroY(2), GyroZ(2), Temperature(1)
+		 */
+		if (drv_data->fifo_data[0] & BIT_FIFO_HEAD_ACCEL) {
+			/* Check empty values */
+			if (!(drv_data->fifo_data[1] == FIFO_ACCEL0_RESET_VALUE
+			      && drv_data->fifo_data[2] ==
+			      FIFO_ACCEL1_RESET_VALUE)) {
+				drv_data->accel_x =
+					((drv_data->fifo_data[1] & 0xff) << 8)
+					+ (drv_data->fifo_data[2] & 0xff);
+				drv_data->accel_y =
+					((drv_data->fifo_data[3] & 0xff) << 8)
+					+ (drv_data->fifo_data[4] & 0xff);
+				drv_data->accel_z =
+					((drv_data->fifo_data[5] & 0xff) << 8)
+					+ (drv_data->fifo_data[6] & 0xff);
+			}
+			if (!(drv_data->fifo_data[0] & BIT_FIFO_HEAD_GYRO)) {
+				drv_data->temp =
+					(int16_t)(drv_data->fifo_data[7]);
+			} else {
+				if (!(drv_data->fifo_data[7] ==
+				      FIFO_GYRO0_RESET_VALUE &&
+				      drv_data->fifo_data[8] ==
+				      FIFO_GYRO1_RESET_VALUE)) {
+					drv_data->gyro_x =
+						((drv_data->fifo_data[7] &
+						  0xff) << 8)
+						+ (drv_data->fifo_data[8] &
+						   0xff);
+					drv_data->gyro_y =
+						((drv_data->fifo_data[9] &
+						  0xff) << 8)
+						+ (drv_data->fifo_data[10] &
+						   0xff);
+					drv_data->gyro_z =
+						((drv_data->fifo_data[11] &
+						  0xff) << 8)
+						+ (drv_data->fifo_data[12] &
+						   0xff);
+				}
+				drv_data->temp =
+					(int16_t)(drv_data->fifo_data[13]);
+			}
+		} else {
+			if (drv_data->fifo_data[0] & BIT_FIFO_HEAD_GYRO) {
+				if (!(drv_data->fifo_data[1] ==
+				      FIFO_GYRO0_RESET_VALUE &&
+				      drv_data->fifo_data[2] ==
+				      FIFO_GYRO1_RESET_VALUE)) {
+					drv_data->gyro_x =
+						((drv_data->fifo_data[1] &
+						  0xff) << 8)
+						+ (drv_data->fifo_data[2] &
+						   0xff);
+					drv_data->gyro_y =
+						((drv_data->fifo_data[3] &
+						  0xff) << 8)
+						+ (drv_data->fifo_data[4] &
+						   0xff);
+					drv_data->gyro_z =
+						((drv_data->fifo_data[5] &
+						  0xff) << 8)
+						+ (drv_data->fifo_data[6] &
+						   0xff);
+				}
+				drv_data->temp =
+					(int16_t)(drv_data->fifo_data[7]);
+			}
+		}
+	}
+
+	return 0;
+}
+
+static int icm42605_attr_set(struct device *dev,
+			     enum sensor_channel chan,
+			     enum sensor_attribute attr,
+			     const struct sensor_value *val)
+{
+	struct icm42605_data *drv_data = dev->data;
+
+	__ASSERT_NO_MSG(val != NULL);
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ:
+		if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+			if (val->val1 > 1000) {
```

Comment:
For SAMPLEING_FREQUENCY, I think current implementation doesn't make confusion even incorrect input.
If val1 is 1001, accel_hz will be 1000 and if val1 is 11 or -100 or any minus value, it will be 12.
(val1 1000 or 12 will be correct value to accel_hz)
And Sampling frequency limitation is from 1000Hz to 12Hz.